### PR TITLE
dts: bindings: pinctrl: place pincfg-node props at root level

### PR DESCRIPTION
**Pull request by gmarull**
Since we can include other binding files at any level (child, grand-child, etc.) it makes no sense to maintain two copies of pinctrl props definitions (pincfg-node/pincfg-node-group). Instead, pincfg-node.yaml defines props at root level, and it is included where needed, either child-binding or grandchild-binding.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

### CODE REVIEW

**Comment by gmarull** [comment]
Pending: discuss wether this applies for treewide change or not. My vote: it is not, since it is a change to a single driver class (pinctrl) which is currently declared as unstable. Impact should be low, since most pinctrl drivers likely reside in-tree.

**Comment by nashif** [comment]
why is this labeled for the TSC? should be discussed in the Arch WG first.

**Comment by mbolivar-nordic** [comment]
> Impact should be low, since most pinctrl drivers likely reside in-tree.

@dkalowsk is this a breaking change for any out of tree SoC pinctrl drivers you may have?

**Comment by dkalowsk** [comment]
@mbolivar-nordic thanks for checking.  Pulled the change in and ran it locally on our tree, I'm not seeing anything obviously broken in some BAT runs, so I think the statement of low risk is accurate.

**Review by galak — CHANGES_REQUESTED** [review]
Add a comment in release notes about the breaking of this.

**Comment by carlescufi** [comment]
Discussed offline with Martí, and we agree that this does not require the treewide label.

**Comment by gmarull** [comment]
> Add a comment in release notes about the breaking of this.

done

File: dts/bindings/pinctrl/atmel,sam0-pinctrl.yaml [hunk 1]
```diff
@@ -105,6 +95,17 @@ child-binding:
   child-binding:
     description: |
       The grandchild nodes group pins that share the same pin configuration.
+
+    include:
+      - name: pincfg-node.yaml
+        property-allowlist:
+          - bias-pull-up
+          - bias-pull-down
+          - drive-open-drain
```
Comment (henrikbrixandersen) [inline dts/bindings/pinctrl/atmel,sam0-pinctrl.yaml:104]: `drive-open-drain` was not included in the existing allow list?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: dts/bindings/pinctrl/atmel,sam0-pinctrl.yaml (hunk 1 above)
Comment (gmarull) [reply dts/bindings/pinctrl/atmel,sam0-pinctrl.yaml]: thanks for catching, fixed

**Review by henrikbrixandersen — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**